### PR TITLE
Add ensureShape API

### PR DESCRIPTION
**Pull request by fengwuyao**
This PR adds the ops for ensureShape() in Core.
We can ensure the input tensor has the same shape as the given shape.
```js
// Construct a 1D tensor with [1, 2, 3, 4]
const x = tf.tensor1d([1, 2, 3, 4]);

// Ensure that x has the same shape with shape: [4]
tf.ensureShape(x, [4]);
```

Fix https://github.com/tensorflow/tfjs/issues/7225

### CODE REVIEW

File: tfjs-core/src/ops/ensure_shape_test.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+describeWithFlags('ensure_shape', ALL_ENVS, () => {
+  it('basic', () => {
+    const x = tf.ones([2, 3]);
+    expect(ensureShape(x, [2, 3])).toEqual(x);
+  });
+
+  it('different shape', () => {
+    const x = tf.ones([2, 3]);
+    expect(() => ensureShape(x, [5, 3])).toThrowError();
```
Comment (Linchenn) [inline tfjs-core/src/ops/ensure_shape_test.ts:30]: We could add a string for toThrowError, might be 'Invalid argument error. Shape of tensor [2, 3] is not compatible with expected shape [5, 3].'

**Review by Linchenn — CHANGES_REQUESTED** [review]

File: tfjs-core/src/ops/ensure_shape.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * Given an input tensor, returns a new tensor with the same values as the
+ * input tensor with shape `shape`.
+ *
+ * The method supports the null value in tensor. It will still check the shapes,
+ * and null is a placeholder.
+ *
+ *
+ * ```js
+ * const x = tf.tensor1d([1, 2, 3, 4]);
+ * const y = tf.tensor1d([1, null, 3, 4]);
```
Comment (Linchenn) [inline tfjs-core/src/ops/ensure_shape.ts:37]: ‘null’ might be in ensureShape's shape argument. The example could be:
```
const y = tf.tensor2d([1, 2, 3, 4], [2,2]);
tf.ensureShape(y, [null, 2]).print();
```

**Review by Linchenn — APPROVED** [review]
LGTM, thanks!

File: tfjs-core/src/ops/ensure_shape_test.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import * as tf from '../index';
+import {ALL_ENVS, describeWithFlags} from '../jasmine_util';
+import {ensureShape} from './ensure_shape';
+
+describeWithFlags('ensure_shape', ALL_ENVS, () => {
```
Comment (chunnienc) [inline tfjs-core/src/ops/ensure_shape_test.ts:22]: Add tests for:
1. shape with nulls
2. shape with different lengths

**Review by chunnienc — CHANGES_REQUESTED** [review]
Please fix the broken converter tests and make sure CI passes, thanks!

File: tfjs-core/src/ops/ensure_shape_test.ts (hunk 3 above)
Comment (chunnienc) [reply tfjs-core/src/ops/ensure_shape_test.ts:22]: and I guess `describeWithFlags('ensureShape', ALL_ENVS ...` is more appropriate for the naming pattern of op tests?

Comment (fengwuyao) [reply tfjs-core/src/ops/ensure_shape_test.ts]: Sg, added the tests.

File: tfjs-core/src/ops/ensure_shape.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC. All Rights Reserved.
```
Comment (chunnienc) [inline tfjs-core/src/ops/ensure_shape.ts:3]: no "All rights reserved"
You can use https://github.com/chunnienc/tfjs-license-fix with glob to add/fix those headers in a batch.

**Review by chunnienc — APPROVED** [review]
LGTM. You can click the resolve button in github to mark/hide the comments you fixed.
Since your PR now has 2 approvals, it's your choice to merge it now or wait for the third one from matt for more comments.

**Pull request merged**